src/Nethermind/Nethermind.Consensus.AuRa/Transactions/TxFilterTxSource.cs

```diff
@@ -33,6 +33,6 @@ public TxFilterTxSource(ITxSource innerSource, ITxPermissionFilter txPermissionF
         }
 
         public IEnumerable<Transaction> GetTransactions(BlockHeader parent, long gasLimit) => 
-            _innerSource.GetTransactions(parent, gasLimit).Where(tx => _txPermissionFilter.IsAllowed(tx, parent));
+            _innerSource.GetTransactions(parent, gasLimit).Where(tx => _txPermissionFilter.IsAllowed(tx, parent, parent.Number + 1));
```

Comment:
is this needed? it was not my suggestion - I was just confirming that the block was parent